### ISSUE TITLE
Clarification: feeding SMILES + AA sequences into T5ProtChem for joint embeddings

### ISSUE THREAD

**Issue by bhawakshi-acog** [tkella47/T5ProtChem#1]
Hi @tkella47, 

First, thanks for open‐sourcing T5ProtChem! I’m using it to compute joint embeddings of protein + substrate pairs, and wanted to confirm the intended input format:

- For **joint‐embedding extraction**, the paper describes an encoder–decoder setup. My understanding is:
  ```python
  enc = tokenizer(smiles, return_tensors="pt")
  dec = tokenizer("<P>" + "<P>".join(aa_seq), return_tensors="pt")
  out = model(
      input_ids=enc.input_ids,
      attention_mask=enc.attention_mask,
      decoder_input_ids=dec.input_ids,
      decoder_attention_mask=dec.attention_mask,
      output_hidden_states=True,
      return_dict=True,
  )
  joint_tokens = out.decoder_hidden_states[-1]

Is that correct?

- Alternatively, for span‐masking/generation tasks, one builds a single prompt string like:

`"Span-Mask: <extra_id_99> CCO <extra_id_98>  Chem-Mask: <extra_id_99> <P>M<P>A<P>T <extra_id_98>"`

and calls `.generate()`. But that isn’t used for embedding extraction, right? 

**Could you please confirm which pattern you recommend for simply pulling out embeddings? Thanks so much!**

**Comment by tkella47** [comment]
Hi there,

Thanks for trying out T5ProtChem! We’re working on fuller docs, but here’s how you can pull out joint embeddings today:

**1. Encoder–Decoder (your snippet)**  
You’re exactly right—you can pass SMILES into the encoder and the protein (prefixed by `<P>` tokens) into the decoder, then grab:

```python
joint_tokens = out.decoder_hidden_states[-1]
```
2. Encoder‑Only concatenation (recommended)
Because the model’s embedding block is shared across SMILES and proteins, you can simply concatenate them into one encoder input:
```
input_str = smiles + "<P>" + "<P>".join(aa_seq)
enc = tokenizer(input_str, return_tensors="pt")
out = model.encoder(
    input_ids=enc.input_ids,
    attention_mask=enc.attention_mask,
    output_hidden_states=True,
    return_dict=True
)
# then e.g. mean‑pool out.last_hidden_state to get a fixed-size vector
```
This method is fast and gives you a single “pair” embedding without needing the decoder.

3. Span‑Masking / .generate()
Those prompts (<extra_id_99>…) are designed for fine‑tuning or generation tasks. We haven’t benchmarked them for embedding quality, so we don’t generally recommend them for simple embedding extraction.
